### ISSUE TITLE
SendCookies and other props are not always passed

### ISSUE THREAD

**Issue by sportnak** [CRAlpha/react-native-wkwebview#43]
When trying to use react-native-wkwebview I noticed that sometimes props fail to get passed down. I noticed that when I set a breakpoint at https://github.com/CRAlpha/react-native-wkwebview/blob/master/ios/RCTWKWebView/RCTWKWebView.m#L57 sendCookies wasn't being set and tracing it further back, the only props set were source and style. 

Here's the initial jsx:
```
    if (!this.state.loadedCookie) {
      return null;
    }

    if (!this.state.loggedIn && this.state.loadedCookie) {
      return (
        <View style={{marginTop: 20}}>
          <Text>Email</Text>
          <TextInput style={inputStyle} onChangeText={this.changeEmail} />
          <Text>Password</Text>
          <TextInput style={inputStyle} onChangeText={this.changePassword} />
          <Button title="Submit" onPress={this.submitLogin}>Submit</Button>
        </View>
      )
    }
    <WKWebView
        source={{url: URL_STRING}}
        ref={ref => { this.web = ref; }}
        sendCookies={true}
        style={{marginTop: 20}}
      />
```

It started working when I added more props:

```
<WKWebView
        source={{url: URL_STRING}}
        onProgress={(progress) => console.log(progress)}
        automaticallyAdjustContentInsets={true}
        ref={ref => { this.web = ref; }}
        sendCookies={true}
        style={{marginTop: 20}}
        automaticallyAdjustContentInsets={true}
        startInLoadingState={true}
      />
```
Additionally, when I turned on hot reloading, and changed the URL_STRING, it would refresh properly with all props set. I'm wondering if someone with more experience contributing to this can help me figure out what's causing this to fail. But as far as I can tell there seems to be some race condition on initial load. 

I'm using react-native-cookies for my cookie manager. I'm running react-native on my mac OS:10.12.2 through the xcode simulator. Thanks

**Comment by insraq** [comment]
I think this is a known issue. An related issue is mentioned in #36 . Basically some props might not be set before the first request is made. We have plans to address this issue (will probably change the signature). However, we are in the process of upgrading to React Native 0.40+ which means our code base is a mess right now. After we finish the work, we will extract the logic from our codebase.

**Comment by sportnak** [comment]
Ah thanks! Yeah I must've missed it because I was searching for sendCookies specifically. I'll look into ways to contribute moving forward. Thanks!

**Comment by petermikitsh** [comment]
In the meantime, you might be able to get away with settings cookies via JavaScript using the `injectedJavaScript` prop or `evaluateJavaScript`.

**Comment by insraq** [comment]
This should've been fixed in npm version 1.3.0
